react_app/src/Clock.js

```diff
@@ -0,0 +1,64 @@
+import React,{Component} from 'react';
+
+const clockTile = {
+  color: 'white',
+  textAlign:'center'
+}
+const clockSpan = {
+  padding : '4px'
+}
+let runner;
+class Clock extends Component{
+  constructor(props) {
+    super(props);
+    this.state ={
+      'time' : this.getCurrentTime()
+    }
+  }
+  getCurrentTime = () =>{
+    const locale = this.props.locale ? this.props.locale : [];
```

Comment:
create all your variables with.     

const {locale, hour12, etc} = this.props